.clang-format

```diff
@@ -1,101 +1,209 @@
 ---
-Language:        Cpp
+Language: Cpp
 AccessModifierOffset: -2
-AlignAfterOpenBracket: AlwaysBreak
-AlignConsecutiveAssignments: false
-AlignConsecutiveDeclarations: true
-AlignConsecutiveMacros: true
-AlignEscapedNewlinesLeft: true
-AlignOperands:   true
+AlignAfterOpenBracket: DontAlign
+AlignConsecutiveMacros: Consecutive
+AlignConsecutiveAssignments: None
+AlignConsecutiveBitFields: None
+AlignConsecutiveDeclarations: None
+AlignEscapedNewlines: Left
+AlignOperands: DontAlign
 AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: true
+AllowAllConstructorInitializersOnNextLine: true
 AllowAllParametersOfDeclarationOnNextLine: false
-AllowShortBlocksOnASingleLine: false
+AllowShortEnumsOnASingleLine: true
+AllowShortBlocksOnASingleLine: Never
 AllowShortCaseLabelsOnASingleLine: false
-AllowShortFunctionsOnASingleLine: Inline
-AllowShortIfStatementsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: None
+AllowShortLambdasOnASingleLine: All
+AllowShortIfStatementsOnASingleLine: Never
 AllowShortLoopsOnASingleLine: false
-AlwaysBreakAfterDefinitionReturnType: TopLevel
-AlwaysBreakAfterReturnType: All
+AlwaysBreakAfterReturnType: TopLevel
 AlwaysBreakBeforeMultilineStrings: false
-AlwaysBreakTemplateDeclarations: true
-BinPackArguments: false
+AlwaysBreakTemplateDeclarations: MultiLine
+
+# Only works wtih macros that do not take parameters
+AttributeMacros:
+ - HSE_ALWAYS_INLINE
+ - HSE_COLD
+ - HSE_CONST
+ - HSE_HOT
+ - HSE_MAYBE_UNUSED
+ - HSE_NOINLINE
+ - HSE_NORETURN
+ - HSE_PACKED
+ - HSE_READ_MOSTLY
+ - HSE_RETURNS_NONNULL
+ - HSE_SENTINEL
+ - HSE_USED
+ - HSE_WARN_UNUSED_RESULT
+ - HSE_WEAK
+
+# Treating function attribute macros that take parameters as type macros
+# helps format function declarations that use them.
+TypenameMacros:
+ - HSE_NONNULL
+ - HSE_PRINTF
+ - HSE_ALIGNED
+StatementAttributeLikeMacros:
+StatementMacros:
+ForEachMacros:
```

Comment:
See more for each macros in the PR that you closed

Reply:
```
ForEachMacros:
 - cds_list_for_each
 - cds_list_for_each_entry_rcu
 - cds_list_for_each_entry_reverse
 - cds_list_for_each_entry_safe
 - cds_list_for_each_entry_safe_from
 - cds_list_for_each_prev
 - cds_list_for_each_prev_safe
 - cds_list_for_each_safe
 - cn_tree_foreach_leaf
 - cn_tree_foreach_leaf_safe
 - cn_tree_foreach_node
 - kmc_zone_foreach
 - list_for_each
 - list_for_each_entry
 - list_for_each_entry_reverse
 - list_for_each_entry_reverse_safe
 - list_for_each_entry_safe
 - rbtree_postorder_for_each_entry_safe
 - s_list_for_each
 - s_list_for_each_entry
 - s_list_for_each_entry_safe
 - s_list_for_each_safe
 - STAILQ_FOREACH
 - TAILQ_FOREACH
 - TAILQ_FOREACH_REVERSE
```

Reply:
coming in next push

Reply:
kmc_slab_foreach